app/src/main/java/dev/ricsarabia/cryptochallenge/data/repos/BitsoRepo.kt

```diff
@@ -0,0 +1,60 @@
+package dev.ricsarabia.cryptochallenge.data.repos
+
+import dev.ricsarabia.cryptochallenge.di.BitsoNetworkingModule
+import dev.ricsarabia.cryptochallenge.domain.*
+
+/**
+ * Created by Ricardo Sarabia on 2021/11/04.
+ * Class to retrieve Bitso data from different sources.
+ */
+class BitsoRepo {
+    suspend fun getBooks(): BooksData {
+        val response = BitsoNetworkingModule.provideBitsoService().getAvailableBooks()
+
+        if (response.body() == null || !response.body()!!.success)
+            return BooksData.Error("Error al obtener los datos")
+
+        val books: List<Book> = response.body()!!.payload.map {
+            Book(
+                it.book,
+                it.book.substringBefore("_"),
+                it.book.substringAfter("_"),
+                "https://cryptoicon-api.vercel.app/api/icon/" + it.book.substringBefore("_")
```

Comment:
Here you can create a constant variable for this URL, and also for your string expressions :D